### ISSUE TITLE
Search is limited to 10 results

### ISSUE THREAD

**Issue by carniz** [elementary/files#741]
As a proof of concept, I created 11 subdirectories under `$HOME/some/subdir` and touched a .foo and .bar file in each of them:
![screenshot from 2018-11-07 16 07 47 2x](https://user-images.githubusercontent.com/1648530/48139723-75e29700-e2a7-11e8-9e9b-8aaf8070f05f.png)

After enabling hidden files I then pressed CTRL+F to search for .foo files, but files just shows the first 10 results:

![cropped screenshot from 2018-11-07 15 58 16 2x](https://user-images.githubusercontent.com/1648530/48139324-82b2bb00-e2a6-11e8-9797-2edfeb730016.png)

My real world use case was to search for files with a certain pattern in my 11GB `src` directory, and even if that might be something that doesn't apply to most users I still find it probable that many users at least have more than 10 files that matches a certain pattern.

To sum it up I think that files needs a better search function that can list all results, preferably also with filters for type, size, modification/access time, etc.

**Comment by jeremypw** [comment]
You are right that search results in each category is limited to 10. As there are four categories (This folder, Below this folder, Bookmarks, Recent), the total maximum number of results is theoretically 40.  I agree may make more sense to allow more results in one category if there are few or no results in another.  But we have to set an upper limit somewhere to avoid an unmanageable number of hits.

I think we need to distinguish *searching* and *filtering* (usually confused).  In the former case, the user has a specific file that they wish to find and they know what it is so in most cases they can enter enough information to reduce the hits to below 10.   In the latter case, the user wants to find (and use) all files meeting a certain criterion.  Files only offers the first function at the moment.

So I think separate issues need to be created - one for the maximum number of hits shown in the pathbar search function and another for the provision of new filter functionality (which would probably be implemented separately from the pathbar).  The former can be fixed relatively easily, the latter will take some design decisions before coding up.

**Comment by jeremypw** [comment]
Yes, I was thinking of something along those lines.  Or clicking on the ellipsis shows the next N results.

**Comment by jeremypw** [comment]
I am going to class this as "Wishlist" as the searchbar does function (more or less) as designed.

**Comment by jeremypw** [comment]
I am working on a possible solution that uses an action bar in the search result window with a "More results" button.  Clicking this incrementally increases the maximum number of results and the depth of search.
